File: python/hps_learn/FlatTupleMaker.py

import ROOT as r
import numpy as n

class FlatTupleMaker :

    def __init__(self, file_name) :
        
        # Open a ROOT file to write the flat ntuple to
        self.root_file = r.TFile(file_name, "RECREATE")
        
        # Create a ROOT TTree used to store the tuples
        self.tree = r.TTree("preprocessed", "Preprocessed Data")

        # Dictionary containing the tuple names and values
        self.variables = dict()

    def add_variable(self, variable_name) : 

        self.variables[variable_name] = n.zeros(1, dtype=float)
        self.tree.Branch(variable_name, self.variables[variable_name], variable_name + "/D")
        
    def set_variable_value(self, variable_name, value) :

        self.variables[variable_name][0] = value

    def fill(self) : 

        self.tree.Fill()

    def close(self):

        self.root_file.Write()
        self.root_file.Close()

File: src/utils/FlatTupleMaker.cxx
/*
 * @file FlatTupleMaker.cxx
 * @author Omar Moreno
 * @date January 18, 2016
 * @brief 
 *
 */

#include <FlatTupleMaker.h>

FlatTupleMaker::FlatTupleMaker(std::string file_name, std::string tree_name) { 

    file = new TFile(file_name.c_str(), "RECREATE");

    tree = new TTree(tree_name.c_str(), tree_name.c_str());   
    
}

FlatTupleMaker::~FlatTupleMaker() { 
    delete file; 
    delete tree; 
}

void FlatTupleMaker::addVariable(std::string variable_name) { 
    
    variables[variable_name] = 0; 
    tree->Branch(variable_name.c_str(), &variables[variable_name], (variable_name + "/D").c_str()); 
}

bool FlatTupleMaker::hasVariable(std::string variable_name) { 
    
    auto search = variables.find(variable_name); 
    if (search != variables.end()) return true; 

    return false; 
}

void FlatTupleMaker::close() { 
    file->Write();
    file->Close(); 
}

